### PR TITLE
Convert CI from Travis.CI to GitHub workflows

### PR DESCRIPTION
**Pull request by christianvuerings**
Travis.ci no longer support open source project, let's move to GitHub workflows instead.

Related PR: https://github.com/airbnb/node-memwatch/pull/15

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ 10, 12, 14, 16, 18 ]
```
Comment (ljharb) [inline .github/workflows/build.yml:16]:
```suggestion
        node: [ 10, 12, 14, 16, 18, 20, 22, 23 ]
```

**Review by ljharb — APPROVED** [review]
LGTM assuming tests pass (we'd have to check [on your fork](https://github.com/christianvuerings/node-memwatch/actions/runs/11726308277), because they won't run here until they're merged)

File: .github/workflows/build.yml (hunk 1 above)
Comment (christianvuerings) [reply .github/workflows/build.yml:16]: @ljharb See https://github.com/christianvuerings/node-memwatch/pull/2 to validate that the tests are passing:
image
https://github.com/christianvuerings/node-memwatch/actions/runs/11726308277/job/32664625808?pr=2

The tests are failing for Node.js 20 - 22 - 23. I created a separate PR to illustrate the failures: https://github.com/christianvuerings/node-memwatch/pull/3 
image

Do you still want me to commit the change and include `20, 22, 23` even though they're failing?

Comment (ljharb) [reply .github/workflows/build.yml:16]: oof, no, that's a separate issue i guess. let's leave them out for now.

**Pull request merged**